app/client/models/DocModel.ts

```diff
@@ -243,6 +249,18 @@ export class DocModel extends Disposable {
       return pagesToShow.filter(p => !hidden(p));
     }));
     this.visibleDocPages = this.autoDispose(ko.computed(() => this.allPages().filter(p => !p.isHidden())));
+    this.leafPageIds = this.autoDispose(ko.computed(() => {
```

Comment:
I think the need for `leafPageIds` (and other changes in this file) can be avoided, since in reality, Grist already knows which page items have children. But the pages/tree code is so convoluted that it's hard to find the right spot to tap into that information. I looked into it, and found a decent spot, I think -- I'll add as comments in the right files.